### PR TITLE
Add ability to change pull policy on install

### PR DESCRIPTION
**Pull request by chris-crone**
Allows setting the image pull policy on install using a `-pull-policy` flag. Note that the flag is ignored if installed in debug or coverage modes.

The default pull policy of `Always` is maintained.

Tested manually using the following command against a clean Kubernetes cluster:
```console
$ ./bin/installer -namespace=compose -etcd-servers=http://compose-etcd-client:2379 -tag=v0.4.16 -skip-liveness-probes=true -pull-policy Never
INFO[0000] Checking installation state                  
INFO[0000] Install image with tag "v0.4.16" in namespace "compose" 
INFO[0000] Api server: image: "docker/kube-compose-dev-api-server:v0.4.16", pullPolicy: "Never" 
INFO[0000] Controller: image: "docker/kube-compose-dev-controller:v0.4.16", pullPolicy: "Never" 

$ kubectl get all -n compose
NAME                                                                  READY   STATUS              RESTARTS   AGE
pod/compose-5b9f95796d-gc7lr                                          0/1     ErrImageNeverPull   0          8s
pod/compose-api-86cccb67cf-xpqmd                                      0/1     ErrImageNeverPull   0          8s
pod/compose-etcd-0000                                                 1/1     Running             0          1m
pod/compose-etcd-0001                                                 1/1     Running             0          59s
pod/compose-etcd-0002                                                 1/1     Running             0          20s
pod/etcd-operator-etcd-operator-etcd-backup-operator-6b58f6899ktz9g   1/1     Running             0          1m
pod/etcd-operator-etcd-operator-etcd-operator-69f56d5647-vvf4c        1/1     Running             0          1m
pod/etcd-operator-etcd-operator-etcd-restore-operator-7b7584d5n6qx8   1/1     Running             0          1m
```

### CODE REVIEW

**Comment by chris-crone** [comment]
I've made this WIP as I need to add at least a unit test.

File: cmd/installer/main.go [hunk 1]
```diff
@@ -44,6 +48,13 @@ func parseOptions(uninstall *bool, options *install.SafeOptions) *rest.Config {
 	}
 	log.SetLevel(loggerLevel)
 
+	switch pp := corev1types.PullPolicy(pullPolicy); pp {
+	case corev1types.PullAlways, corev1types.PullNever, corev1types.PullIfNotPresent:
+		options.OptionsCommon.PullPolicy = pp
+	default:
+		panic(fmt.Sprintf("invalid pull policy: %q", pullPolicy))
```
Comment (silvin-lubecki) [inline cmd/installer/main.go]: Keep calm and don't panic 😄 
Return an error instead?

Comment (chris-crone) [reply cmd/installer/main.go]: We panic if the log level or etcd certs are incorrect so I did the same for the pull policy. I agree that panic is overkill and we should return an error but we can do that for everything in a followup.

**Comment by chris-crone** [comment]
@silvin-lubecki @simonferquel Refactored and added unit test.

**Pull request merged**